### PR TITLE
Clone a stack without using extra space

### PR DESCRIPTION
**Pull request by Ratnesh4193**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Comment by Ratnesh4193** [comment]
this pr is for #4028 issue.
kindly review this and merge it

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "files.associations": {
+        "array": "cpp",
+        "any": "cpp",
+        "string_view": "cpp",
+        "regex": "cpp"
+    }
+}
```
Comment (HarshCasper) [inline .vscode/settings.json]: Remove this

File: C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	}
+}
+int main()
+{
+	ios_base::sync_with_stdio(false); cin.tie(NULL);
+
+#ifndef ONLINE_JUDGE
+	freopen("input1.txt", "r", stdin);
+	freopen("error.txt", "w", stderr);
+	freopen("output.txt", "w", stdout);
```
Comment (HarshCasper) [inline C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp]: Why do we have these files? We are not having any online judges

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by Ratnesh4193 — COMMENTED** [review]
MADE NECESSARY CHANGES

File: C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
+// Below is the implementation of
+// Cloning a stack into destination stack without using extra space
+
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp]: Remove extra space and write bit more description.

File: C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+#include <bits/stdc++.h>
+using namespace std;
+typedef unsigned long int ul;
+typedef long long int ll;
+typedef unsigned long long int  ull;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp]: Remove CP templates and tricks

File: C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp]: Remove extra space and write function description

File: C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+	{
+		int x = st.top();
+		st.pop();
+
+		clonestack(st, cloned);
+
+		cloned.push(x);
+	}
+}
+int main()
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp:23]: Add a new line

File: C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+		cout << source.top() << " ";
+		source.pop();
+	}
+	cout << endl;
+	cout << "Cloned Stack : ";
+	while (!dest.empty()) {
+		cout << dest.top() << " ";
+		dest.pop();
+	}
+	// cerr << "time taken : " << (float)clock() / CLOCKS_PER_SEC << " secs" << endl;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp]: Remove redundant comments

File: C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+	}
+}
+int main()
+{
+	
+	stack<int> source, dest;
+	source.push(1);
+	source.push(2);
+	source.push(3);
+	source.push(4);
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp]: Use dynamic input from user

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Ratnesh4193** [comment]
@ankitaggarwal23 what changes I have to do now??

File: C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp [hunk 9]
```diff
@@ -0,0 +1,55 @@
+// Below is the implementation of
+// Cloning a stack into destination stack without using extra space
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp]: Use multiline comments and write proper description

File: C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+#include<bits/stdc++.h>
+using namespace std;
+// This Function takes takes two stakes as input and
+// copy data of first stack to the second stack  
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp]: Remove the redundant comments instead write one line about your function description

File: C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+void clonestack(stack<int> st, stack<int> &cloned) 
+{
+	if (st.size() > 0) //this is the base case when the source stack is empty
+	{
+		int x = st.top();  //stores the top of the stack
+		st.pop();
+
+		clonestack(st, cloned); //this will copy element of rest of stack
+
+		cloned.push(x); // now put the current element on the top
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp]: Remove the redundant comments

File: C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+	}
+}
+int main()
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp:23]: Add a new line above the main

File: C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+	return 0;
+}
+
+/*
+
+		OUTPUT
+
+	Source Stack : 4 3 2 1 
+	
+	Cloned Stack : 4 3 2 1
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp]: Write the output in the way your code is producing and remove the tab spaces from the start Also write TC and SC

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Ratnesh4193** [comment]
Kindly review @ankitaggarwal23 

OUTPUT SCREENSHOT
Capture

File: C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp [hunk 14]
```diff
@@ -0,0 +1,58 @@
+/* Below is the implementation of Cloning a stack into destination stack without using extra space */
+#include<bits/stdc++.h>
+using namespace std;
+/* This Function takes takes two stakes as input and copy data of first stack to the second stack.
+   It is a recursive code in which we store the top element of the stack in a temporary variable
+   and call the function which clones rest of the stack and push the stored to the top of
+   the stack.
+*/
+void clonestack(stack<int> st, stack<int> &cloned)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp]: Rewrite it as,

```
/* 
Cloning stack without extra space:

This Function takes takes two stakes as input and copy data of first stack to the second stack.
It is a recursive code in which we store the top element of the stack in a temporary variable
and call the function which clones rest of the stack and push the stored to the top of
the stack.

*/

#include<bits/stdc++.h>
using namespace std;

void clonestack(stack<int> st, stack<int> &cloned)
```

File: C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+	5
+	Enter the elements of stack
+	3 5 1 2 4
+
+	OUTPUT
+
+	Source Stack : 3 5 1 2 4
+	Cloned Stack : 3 5 1 2 4
+
+	*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp]: Left align it properly

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Ratnesh4193** [comment]
made changes kindly review @ankitaggarwal23

File: C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+		Enter the elements of stack
+		3 5 1 2 4
+
+	OUTPUT: Source Stack : 3 5 1 2 4
+		Cloned Stack : 3 5 1 2 4
+
+	Time and space complexity for this approach:
+	  Time Complexity: O(N)
+	  Auxiliary Space: O(1)
+	*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp]: Left align it we are telling you in every review.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Also check your code now showing error.

File: C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp (hunk 16 above)
Comment (ankitaggarwal23) [reply C-Plus-Plus/stack/clone_a_stack_without_usinig_extra_space.cpp]: Also resolve merge conflict

**Review by ankitaggarwal23 — COMMENTED** [review]
Last Reminder, Are you still working on it.

**Review by ankitaggarwal23 — COMMENTED** [review]
Last Reminder, Are you still working on it.

**Comment by Ratnesh4193** [comment]
@ankitaggarwal23  Sorry I was inactive for the past few days.
I will check and remove the error by evening.
I don't know how to solve merge conflict.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**